clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java

```diff
@@ -145,7 +155,20 @@ private SSLContext createSSLContext() throws GeneralSecurityException, IOExcepti
         KeyStore ts = truststore == null ? null : truststore.load();
         tmf.init(ts);
 
-        sslContext.init(keyManagers, tmf.getTrustManagers(), this.secureRandomImplementation);
+        if ((wantClientAuth || needClientAuth) && truststore != null && mode == Mode.SERVER) {
```

Comment:
In case of Kafka clients, truststore reloading is not that useful since this feature does not help after connection is established. Also restarting a Kafka client is much more trivial than restarting a Kafka broker cluster.